reference/mysql-compatibility.md

```diff
@@ -247,6 +247,12 @@ Because they are built-in, named time zones in TiDB might behave slightly differ
 
 It is not recommended to unset the `NO_ZERO_DATE` and `NO_ZERO_IN_DATE` SQL modes, which are enabled by default in TiDB as in MySQL. While TiDB supports operating with these modes disabled, the TiKV coprocessor does not. Executing certain statements that push down date and time processing functions to TiKV might result in a statement error.
 
+#### Handling of space at the end of string type line
+
+Currently, when inserting data, TiDB will keep the space at the end of the line for the `VARCHAR` type, and truncate the space for the `CHAR` type. In case there is no index, TiDB behaves exactly the same as MySQL. If there is a `UNIQUE` index on the `VARCHAR` type, MySQL will truncate the space at the end of the `VARCHAR` data before determining whether the data is duplicated, which is similar to the processing of the `CHAR` type, while TiDB will keep the space.
```

Comment:
```suggestion
Currently, when inserting data, TiDB keeps the space at the end of the line for the `VARCHAR` type, and truncate the space for the `CHAR` type. In case there is no index, TiDB behaves exactly the same as MySQL. 

If there is a `UNIQUE` index on the `VARCHAR` data, MySQL truncates the space at the end of the `VARCHAR` line before determining whether the data is duplicated, which is similar to the processing of the `CHAR` type, while TiDB keeps the space.
```